src/com/carolinarollergirls/scoreboard/core/impl/BoxTripImpl.java

```diff
@@ -231,7 +231,7 @@ public void execute(CommandProperty prop) {
     public void end() {
         set(Value.IS_CURRENT, false);
         set(Value.WALLTIME_END, ScoreBoardClock.getInstance().getCurrentWalltime());
-        if (getTeam().hasFieldingAdvancePending()) {
+        if (getTeam().hasFieldingAdvancePending() && getCurrentFielding().getTeamJam().isRunningOrUpcoming()) {
```

Comment:
There's far more broken than this, and this needs unittests. You should start with the ones I wrote, that covers all the corner cases I found.

Reply:
Instead of vaguely alluding to stuff that's allegedly broken, you should give specific bug reports. And if you want unit tests, provide some that test the intended functionality and not some that test for a feature to be missing.

Reply:
The tests I wrote are pretty extensive. It's so broken as is, that 2-3 clicks is enough to show a bug when editing a box trip.

Reply:
Put another way, run those unittests by hand and see if they do what you expect. The removal of that behaviour was incidental and was just the way that PR worked out, my goal was to fix the general broken box trip editor.

Reply:
"Your code is a pile of shit" is not exactly motivating me to put any effort into following up. I will not do any work on this unless I get specific bug reports (that are reproducible with the changes from this PR applied). (And if I do change anything, it would be in a separate PR.)

Reply:
It's a complex thing to fix so doing it in another PR is fine, though this one should not be marked as fixing #269 as it doesn't.

Reply:
Actually it does. #269 is about adding a skater and a box trip in the same go which was broken as the former was delayed and the latter wasn't. This PR makes adding Skaters immediate, so it addresses this specific scenario. If you have other scenarios, report them properly.

Reply:
https://github.com/rollerderby/scoreboard/issues/269#issuecomment-500606708 covers another. And the unittests cover the rest.

Reply:
I don't have the faintest idea what that comment is talking about. But if you insist on keeping that thing open, so be it. I'd rather get this functionality in.

Reply:
Create a penalty in the previous jam, advance the start time forward. It'll say Jam 0 at some point.

Considering it's a bugfix, can you add a unittest for this particular change? (Or leave it all for another PR)

Reply:
Ok, I can reproduce that. I have an idea what the cause is, but that's definitely it's own PR.

Everything else should be addressed.

Reply:
That's just one form of it, my PR handles all of them so I'd suggest taking a look as I've done most of the leg work already.